stingray/simulator/simulator.py

```diff
@@ -134,13 +132,13 @@ def simulate(self, *args):
             lightCurve: `LightCurve` object
         """
 
-        if type(args[0]) == int:
+        if type(args[0]) == int and len(args) == 1: 
             return  self._simulate_power_law(args[0])
```

Comment:
Type comparison might not work if instead of an integer you have a long. This should be more generic:

```
import numbers
(...)
if isinstance(args[0], numbers.Integral):
    (...)
```